fix, on ppc64le and RHEL / CentOS 8, for: identifier "__ieee128" is undefined
See https://github.com/easybuilders/easybuild-easyconfigs/issues/11913
Patch by Simon Branford (University of Birmingham), ported to PyTorch 1.7.1 by Kenneth Hoste (HPC-UGent)
--- pytorch-1.7.1/cmake/Dependencies.cmake.orig	2020-12-07 19:28:38.000000000 +0000
+++ pytorch-1.7.1/cmake/Dependencies.cmake	2021-01-10 14:09:00.593547793 +0000
@@ -1501,6 +1501,9 @@
   if(CMAKE_POSITION_INDEPENDENT_CODE AND NOT MSVC)
     list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "-fPIC")
   endif()
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "(powerpc|ppc)64le")
+    list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "-mno-float128")
+  endif()
 
   if(CUDA_HAS_FP16 OR NOT ${CUDA_VERSION} LESS 7.5)
     message(STATUS "Found CUDA with FP16 support, compiling with torch.cuda.HalfTensor")
--- pytorch-1.7.1/third_party/gloo/cmake/Cuda.cmake.orig	2021-01-10 14:09:00.593547793 +0000
+++ pytorch-1.7.1/third_party/gloo/cmake/Cuda.cmake	2021-01-10 14:11:36.177738536 +0000
@@ -177,5 +177,9 @@
   gloo_list_append_if_unique(CUDA_NVCC_FLAGS "-Xcompiler" "-fPIC")
 endif()
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "(powerpc|ppc)64le")
+  gloo_list_append_if_unique(CUDA_NVCC_FLAGS "-Xcompiler" "-mno-float128")
+endif()
+
 mark_as_advanced(CUDA_BUILD_CUBIN CUDA_BUILD_EMULATION CUDA_VERBOSE_BUILD)
 mark_as_advanced(CUDA_SDK_ROOT_DIR CUDA_SEPARABLE_COMPILATION)
